ksql-engine/src/test/java/io/confluent/ksql/physical/PhysicalPlanBuilderTest.java

```diff
@@ -250,83 +251,107 @@ public void shouldCreateExecutionPlanForInsert() {
     assertThat(ksqlStructuredDataOutputNode.getKsqlTopic().getKsqlTopicSerDe().getSerDe(),
         equalTo(DataSource.DataSourceSerDe.DELIMITED));
     closeQueries(queryMetadataList);
-    ksqlEngine.close();
   }
 
   @Test
   public void shouldFailIfInsertSinkDoesNotExist() {
+    // Given:
     final String insertIntoQuery = "INSERT INTO s1 SELECT col0, col1, col2 FROM test1;";
+    kafkaTopicClient.createTopic("test1", 1, (short) 1, Collections.emptyMap());
 
+    // Then:
     expectedException.expect(KsqlStatementException.class);
     expectedException.expect(statementText(is("INSERT INTO s1 SELECT col0, col1, col2 FROM test1;")));
-    expectedException
-        .expect(rawMessage(is("Sink 'S1' does not exist for the INSERT INTO statement.")));
+    expectedException.expect(rawMessage(is(
+        "Sink 'S1' does not exist for the INSERT INTO statement.")));
 
-    try {
-      KsqlEngineTestUtil.execute(
+    // When:
+    KsqlEngineTestUtil.execute(
           ksqlEngine,
           createStream + "\n " + insertIntoQuery,
           ksqlConfig,
           Collections.emptyMap());
-    } finally {
-      ksqlEngine.close();
-    }
   }
 
   @Test
   public void shouldFailInsertIfTheResultSchemaDoesNotMatch() {
+    // Given:
     final String csasQuery = "CREATE STREAM s1 AS SELECT col0, col1 FROM test1;";
     final String insertIntoQuery = "INSERT INTO s1 SELECT col0, col1, col2 FROM test1;";
     kafkaTopicClient.createTopic("test1", 1, (short) 1, Collections.emptyMap());
 
-    try {
-      KsqlEngineTestUtil.execute(
+    // Then:
+    expectedException.expect(KsqlStatementException.class);
+    expectedException.expect(rawMessage(is(
+        "Incompatible schema between results and sink. Result schema is "
+        + "[COL0 : BIGINT, COL1 : VARCHAR, COL2 : DOUBLE], "
+        + "but the sink schema is [COL0 : BIGINT, COL1 : VARCHAR].")));
+
+    // When:
+    KsqlEngineTestUtil.execute(
           ksqlEngine,
           createStream + "\n " + csasQuery + "\n " + insertIntoQuery,
           ksqlConfig,
           Collections.emptyMap());
-    } catch (final KsqlException ksqlException) {
-      assertThat(ksqlException.getMessage(),
-          equalTo("Incompatible schema between results and sink. Result schema is [COL0 :"
-              + " BIGINT, COL1 : VARCHAR, COL2 : DOUBLE], but the sink schema is [COL0 : BIGINT, COL1 : VARCHAR]."));
-      return;
-    } finally {
-      ksqlEngine.close();
-    }
-    Assert.fail();
   }
 
   @Test
-  public void shouldCreatePlanForInsertIntoTableFromTable() {
+  public void shouldThrowOnInsertIntoTableFromTable() {
```

Comment:
Old test was invalid.  The statements should of thrown, but the `parseStatement` call registers `T2` as a stream, not a table, so the test passes.